frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -175,22 +182,79 @@ ListDropdown.propTypes = {
   id: PropTypes.string,
 };
 
-const NsDropdown_ = (props) => {
-  const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
-  if (flagPending(openshiftFlag)) {
-    return null;
+const useProjectOrNamespaceModel = () => {
+  const nsModel = modelFor('Namespace');
+  const canCreateNamespace = useAccessReview({
+    group: nsModel.apiGroup,
+    resource: nsModel.plural,
+    verb: 'create',
+  });
+
+  const canCreateProject = useFlag(FLAGS.CAN_CREATE_PROJECT);
+  const openshiftFlag = useFlag(FLAGS.OPENSHIFT);
+
+  if (flagPending(openshiftFlag) || flagPending(canCreateProject)) {
+    return [];
+  }
+
+  if (canCreateNamespace) {
+    return [modelFor('Namespace'), canCreateNamespace];
+  } else if (canCreateProject) {
+    return [modelFor('Project'), canCreateProject];
   }
```

Comment:
We always need to return `[model, canCreate]` from this function once we have the flags. The model is used to list the resources whether or not the user can create them. The `openshiftFlag` tells us whether or not we are running against an OpenShift cluster. If it's true, we need to use the OpenShift-specific `ProjectModel`, otherwise default to the vanilla K8s`NamespaceModel`. Also, import and use static models instead of retrieving them dynamically:

```suggestion
  return openshiftFlag 
    ? [ProjectModel, canCreateProject] 
    : [NamespaceModel, canCreateNamespace];
```